CHANGELOG.md

```diff
@@ -11,6 +11,7 @@ These changes are available in the [master branch](https://github.com/PrefectHQ/
 ### Enhancements
 
 - Improve error handling for unsupported callables - [#1993](https://github.com/PrefectHQ/prefect/pull/1993)
+- Ensure all calls to `flow.run()` use the same execution logic - [#1994](https://github.com/PrefectHQ/prefect/pull/1994)
```

Comment:
Isn't this also a breaking change for `flow.run` now that any retries will actually be respected instead of returned?

Reply:
Certainly new behavior, but is it breaking?

Reply:
I'd call it breaking - anyone who uses `flow.run()` currently will have things like `Retrying` states immediately returned.  With the proposed changes this call will block until all of the retries have occurred.  I don't think it's a _massive_ deal but given that this is the main entrypoint for open source users I'm a bit nervous about it.

Reply:
Sorry this has sat for a while - would you like me just to move this to the `breaking` section of the changelog, or take a larger step to introduce the behavior?

Reply:
@jlowin yea could you add it to the breaking changes section?